# DAT210x - Programming with Python for DS

## Module2 - Lab2

In [2]:
# Import and alias Pandas
import pandas as pd

Write code below to load up the `tutorial.csv` dataset. You can store it into a variable called df:

In [3]:
df=pd.read_csv('Datasets/tutorial.csv')
print(df)

       col0      col1      col2      col3
0 -0.722876 -1.330682  1.309208  0.232378
1  1.160396 -0.730879  0.677368  1.044722
2 -1.062870 -0.503704 -0.238536 -1.417937
3  0.437078  0.362640 -0.111228 -1.649853


Now that your dataset has been loaded, invoke the `.describe()` method to display some results about it:

In [4]:
df.describe()

col0      col1      col2      col3
count  4.000000  4.000000  4.000000  4.000000
mean  -0.047068 -0.550656  0.409203 -0.447672
std    1.029715  0.701720  0.723956  1.300815
min   -1.062870 -1.330682 -0.238536 -1.649853
25%   -0.807875 -0.880830 -0.143055 -1.475916
50%   -0.142899 -0.617291  0.283070 -0.592780
75%    0.617907 -0.287118  0.835328  0.435464
max    1.160396  0.362640  1.309208  1.044722

Lastly, try experimenting with indexing. Figure out which _indexing method_ you would need to use in order to index your dataframe with: `[2:4, 'col3']`. Finally, display the results:

In [10]:
df.loc[2:4,['col3']]

col3
2 -1.417937
3 -1.649853

In [12]:
dff = pd.read_csv('Datasets/direct_marketing.csv')

In [13]:
dff

recency   history_segment  history  mens  womens   zip_code  newbie  \
0           10    2) $100 - $200   142.44     1       0  Surburban       0   
1            6    3) $200 - $350   329.08     1       1      Rural       1   
2            7    2) $100 - $200   180.65     0       1  Surburban       1   
3            9    5) $500 - $750   675.83     1       0      Rural       1   
4            2      1) $0 - $100    45.34     1       0      Urban       0   
5            6    2) $100 - $200   134.83     0       1  Surburban       0   
6            9    3) $200 - $350   280.20     1       0  Surburban       1   
7            9      1) $0 - $100    46.42     0       1      Urban       0   
8            9    5) $500 - $750   675.07     1       1      Rural       1   
9           10      1) $0 - $100    32.84     0       1      Urban       1   
10           7    5) $500 - $750   548.91     0       1      Urban       1   
11           1    3) $200 - $350   211.45     0       1      Urban       1   
12           5    5) $500 - $750   642.90     0       1  Surburban       1   
13           2    2) $100 - $200   101.64     0       1      Urban       0   
14           4    3) $200 - $350   241.42     0       1      Rural       1   
15           3      1) $0 - $100    58.13     1       0      Urban       1   
16           5      1) $0 - $100    29.99     1       0  Surburban       0   
17           9    2) $100 - $200   112.35     1       0      Rural       0   
18          11    3) $200 - $350   219.04     1       1  Surburban       0   
19           5  6) $750 - $1,000   828.42     1       0  Surburban       1   
20           9      1) $0 - $100    29.99     0       1  Surburban       1   
21          11    2) $100 - $200   182.32     1       0  Surburban       0   
22           2    2) $100 - $200   118.40     1       0  Surburban       0   
23           2      1) $0 - $100    29.99     0       1      Urban       1   
24           4      1) $0 - $100    78.24     1       0  Surburban       0   
25           6    2) $100 - $200   139.87     0       1      Rural       1   
26          12    2) $100 - $200   115.39     0       1  Surburban       0   
27           6    2) $100 - $200   162.98     0       1  Surburban       0   
28           7    4) $350 - $500   435.73     0       1      Urban       1   
29           2    3) $200 - $350   203.35     1       0      Rural       0   
...        ...               ...      ...   ...     ...        ...     ...   
63970        4    2) $100 - $200   191.15     0       1  Surburban       1   
63971        5    5) $500 - $750   549.87     0       1  Surburban       1   
63972        3    5) $500 - $750   554.97     0       1  Surburban       1   
63973        8    4) $350 - $500   471.80     1       0  Surburban       1   
63974        8      1) $0 - $100    73.65     0       1  Surburban       0   
63975       10  6) $750 - $1,000   883.92     1       0      Urban       1   
63976        1    5) $500 - $750   710.72     1       1      Urban       1   
63977        8      1) $0 - $100    29.99     0       1  Surburban       0   
63978       10    2) $100 - $200   102.01     0       1      Urban       1   
63979       10    2) $100 - $200   168.21     0       1  Surburban       0   
63980        3    4) $350 - $500   487.10     0       1  Surburban       1   
63981        4    2) $100 - $200   125.53     0       1      Rural       1   
63982        5      1) $0 - $100    29.99     1       0      Urban       1   
63983        2      1) $0 - $100    83.03     0       1      Urban       0   
63984        2    3) $200 - $350   209.51     0       1      Urban       1   
63985        9      1) $0 - $100    29.99     1       0      Urban       0   
63986        9      1) $0 - $100    35.26     0       1      Urban       1   
63987        1      1) $0 - $100    79.70     1       0  Surburban       1   
63988        6      1) $0 - $100    32.98     1       0  Surburban       0   
63989       10    3) $200 - $

In [16]:
dff[['recency']]

recency
0           10
1            6
2            7
3            9
4            2
5            6
6            9
7            9
8            9
9           10
10           7
11           1
12           5
13           2
14           4
15           3
16           5
17           9
18          11
19           5
20           9
21          11
22           2
23           2
24           4
25           6
26          12
27           6
28           7
29           2
...        ...
63970        4
63971        5
63972        3
63973        8
63974        8
63975       10
63976        1
63977        8
63978       10
63979       10
63980        3
63981        4
63982        5
63983        2
63984        2
63985        9
63986        9
63987        1
63988        6
63989       10
63990        6
63991        1
63992        1
63993        4
63994        7
63995       10
63996        5
63997        6
63998        1
63999        1

[64000 rows x 1 columns]

In [25]:
dff.iloc[:, [1,3]]

history_segment  mens
0        2) $100 - $200     1
1        3) $200 - $350     1
2        2) $100 - $200     0
3        5) $500 - $750     1
4          1) $0 - $100     1
5        2) $100 - $200     0
6        3) $200 - $350     1
7          1) $0 - $100     0
8        5) $500 - $750     1
9          1) $0 - $100     0
10       5) $500 - $750     0
11       3) $200 - $350     0
12       5) $500 - $750     0
13       2) $100 - $200     0
14       3) $200 - $350     0
15         1) $0 - $100     1
16         1) $0 - $100     1
17       2) $100 - $200     1
18       3) $200 - $350     1
19     6) $750 - $1,000     1
20         1) $0 - $100     0
21       2) $100 - $200     1
22       2) $100 - $200     1
23         1) $0 - $100     0
24         1) $0 - $100     1
25       2) $100 - $200     0
26       2) $100 - $200     0
27       2) $100 - $200     0
28       4) $350 - $500     0
29       3) $200 - $350     1
...                 ...   ...
63970    2) $100 - $200     0
63971    5) $500 - $750     0
63972    5) $500 - $750     0
63973    4) $350 - $500     1
63974      1) $0 - $100     0
63975  6) $750 - $1,000     1
63976    5) $500 - $750     1
63977      1) $0 - $100     0
63978    2) $100 - $200     0
63979    2) $100 - $200     0
63980    4) $350 - $500     0
63981    2) $100 - $200     0
63982      1) $0 - $100     1
63983      1) $0 - $100     0
63984    3) $200 - $350     0
63985      1) $0 - $100     1
63986      1) $0 - $100     0
63987      1) $0 - $100     1
63988      1) $0 - $100     1
63989    3) $200 - $350     1
63990      1) $0 - $100     0
63991    3) $200 - $350     1
63992    5) $500 - $750     1
63993    4) $350 - $500     0
63994      1) $0 - $100     0
63995    2) $100 - $200     1
63996      1) $0 - $100     0
63997      1) $0 - $100     1
63998    5) $500 - $750     1
63999    4) $350 - $500     0

[64000 rows x 2 columns]

In [28]:
dff[dff.recency < 7]

recency   history_segment  history  mens  womens   zip_code  newbie  \
1            6    3) $200 - $350   329.08     1       1      Rural       1   
4            2      1) $0 - $100    45.34     1       0      Urban       0   
5            6    2) $100 - $200   134.83     0       1  Surburban       0   
11           1    3) $200 - $350   211.45     0       1      Urban       1   
12           5    5) $500 - $750   642.90     0       1  Surburban       1   
13           2    2) $100 - $200   101.64     0       1      Urban       0   
14           4    3) $200 - $350   241.42     0       1      Rural       1   
15           3      1) $0 - $100    58.13     1       0      Urban       1   
16           5      1) $0 - $100    29.99     1       0  Surburban       0   
19           5  6) $750 - $1,000   828.42     1       0  Surburban       1   
22           2    2) $100 - $200   118.40     1       0  Surburban       0   
23           2      1) $0 - $100    29.99     0       1      Urban       1   
24           4      1) $0 - $100    78.24     1       0  Surburban       0   
25           6    2) $100 - $200   139.87     0       1      Rural       1   
27           6    2) $100 - $200   162.98     0       1  Surburban       0   
29           2    3) $200 - $350   203.35     1       0      Rural       0   
30           2    3) $200 - $350   237.53     0       1  Surburban       0   
32           6    2) $100 - $200   128.01     0       1      Urban       0   
34           3      1) $0 - $100    29.99     1       0      Rural       0   
35           4    3) $200 - $350   218.72     0       1      Urban       0   
36           1    5) $500 - $750   514.52     0       1  Surburban       1   
37           4  6) $750 - $1,000   766.47     1       1      Urban       1   
41           3      1) $0 - $100    99.23     1       0      Rural       0   
43           2    4) $350 - $500   492.02     1       0  Surburban       0   
44           1      1) $0 - $100    48.32     0       1      Urban       0   
46           2    4) $350 - $500   391.33     1       0  Surburban       0   
47           1    5) $500 - $750   729.70     1       1  Surburban       1   
48           3    2) $100 - $200   134.59     1       0      Urban       1   
50           3    3) $200 - $350   203.30     0       1  Surburban       0   
55           6      1) $0 - $100    42.66     1       0  Surburban       0   
...        ...               ...      ...   ...     ...        ...     ...   
63949        5      1) $0 - $100    86.79     1       0      Rural       0   
63950        1      1) $0 - $100    45.67     0       1  Surburban       1   
63953        5    2) $100 - $200   166.24     0       1      Urban       0   
63954        2      1) $0 - $100    93.97     1       0      Urban       0   
63955        1      1) $0 - $100    29.99     1       0  Surburban       0   
63960        1    3) $200 - $350   221.89     0       1  Surburban       1   
63961        4    3) $200 - $350   337.36     1       0      Urban       0   
63964        2  6) $750 - $1,000   772.99     1       1  Surburban       1   
63966        4    2) $100 - $200   170.03     1       0  Surburban       0   
63967        5      1) $0 - $100    77.73     0       1      Urban       1   
63969        3      1) $0 - $100    67.78     0       1  Surburban       0   
63970        4    2) $100 - $200   191.15     0       1  Surburban       1   
63971        5    5) $500 - $750   549.87     0       1  Surburban       1   
63972        3    5) $500 - $750   554.97     0       1  Surburban       1   
63976        1    5) $500 - $750   710.72     1       1      Urban       1   
63980        3    4) $350 - $500   487.10     0       1  Surburban       1   
63981        4    2) $100 - $200   125.53     0       1      Rural       1   
63982        5      1) $0 - $100    29.99     1       0      Urban       1   
63983        2      1) $0 - $100    83.03     0       1      Urban       0   
63984        2    3) $200 - $

### Lab Question 1

When you print the results of calling .describe() on your dataframe, what is the value displayed in the bottom right corner (col3 max)?

#### Answer:
1.044722

### Lab Question 2

Which of the many indexing methods did you use to get [2:4,'col3'] working?

#### Answer:
df.loc

### Lab Question 3

How many values are returned when you print the results of the [2:4,'col3'] indexing operation?

#### Answer:
2